### ISSUE TITLE
Add Mutual Information Scoring Utility to Preprocessing Module

### ISSUE THREAD

**Issue by idanmoradarthas** [idanmoradarthas/DataScienceUtils#64]
### Description

We want to add a utility function in `ds_utils.preprocess` to **calculate mutual information scores** between features and a target label. This will assist in feature selection by quantifying the dependency between input features and the label, supporting both **numerical and categorical** inputs.

The function should:
- Automatically detect column types (numeric, boolean, categorical).
- Preprocess each type accordingly (imputation).
- Handle categorical target encoding.
- Return a sorted DataFrame of features and their MI scores.

---

### 🧪 Reference Code

```python
import pandas as pd
import numpy as np
from sklearn.feature_selection import mutual_info_classif
from sklearn.impute import SimpleImputer
from sklearn.preprocessing import OneHotEncoder, LabelEncoder
from sklearn.compose import ColumnTransformer
from sklearn.pipeline import Pipeline

def compute_mutual_information(link_count_features, features, label_col='label'):
    feature_name_label = 'feature_name'
    mi_label = 'mi_score'

    # A) Identify feature types
    numerical_features = link_count_features[features].select_dtypes(include=np.number).columns.tolist()
    boolean_features = link_count_features[features].select_dtypes(include=bool).columns.tolist()
    categorical_features = link_count_features[features].select_dtypes(include=['object', 'category']).columns.tolist()

    # B) Preprocessing pipelines
    numerical_transformer = Pipeline(steps=[('imputer', SimpleImputer(strategy='mean'))])
    discrete_transformer = Pipeline(steps=[('imputer', SimpleImputer(strategy='most_frequent'))])

    preprocessor = ColumnTransformer(
        transformers=[
            ('num', numerical_transformer, numerical_features),
            ('discrete', discrete_transformer, boolean_features + categorical_features)
        ],
        remainder='passthrough'
    )

    discrete_features_mask = [False] * len(numerical_features) + [True] * len(boolean_features + categorical_features)

    # C) Apply preprocessing
    X_preprocessed = preprocessor.fit_transform(link_count_features[features])

    # D) Encode target
    le = LabelEncoder()
    y_encoded = le.fit_transform(link_count_features[label_col])

    # E) Compute MI
    mi_score = mutual_info_classif(
        X=X_preprocessed,
        y=y_encoded,
        random_state=42,
        discrete_features=discrete_features_mask
    )

    final_feature_names = numerical_features + boolean_features + categorical_features

    mi_df = pd.DataFrame({
        feature_name_label: final_feature_names,
        mi_label: mi_score
    })

    return mi_df.sort_values(by=[mi_label], ascending=False)
```
### ✅ Acceptance Criteria

- [ ] A new function `compute_mutual_information(df, features, label_col='label')` is added to the `ds_utils.preprocess` module.
- [ ] The function automatically detects feature types:
  - Numerical
  - Boolean
  - Categorical (object or category dtype)
- [ ] Feature preprocessing includes:
  - Mean imputation for numerical features
  - Most frequent imputation for boolean and categorical features
- [ ] The target label column is encoded using `LabelEncoder` if not already numeric.
- [ ] The function computes mutual information scores using `mutual_info_classif`.
- [ ] The result is a `pandas.DataFrame` with columns:
  - `feature_name`
  - `mi_score`
  - Sorted by `mi_score` in descending order
- [ ] Unit tests validate correct handling of feature types, missing values, and score calculation.
- [ ] Docstring is included with detailed argument and return descriptions.
- [ ] Usage example added to the README.

---

### 📁 Related File

```
ds_utils/preprocess/compute_mutual_information
```
